analyzers/src/SonarAnalyzer.VisualBasic/Rules/InsteadOfAny.cs

```diff
@@ -21,15 +21,12 @@
 namespace SonarAnalyzer.Rules.VisualBasic;
 
 [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
-public sealed class ExistsInsteadOfAny : ExistsInsteadOfAnyBase<SyntaxKind, InvocationExpressionSyntax>
+public sealed class InsteadOfAny : InsteadOfAnyBase<SyntaxKind, InvocationExpressionSyntax>
 {
     protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
 
-    protected override bool HasOneArgument(InvocationExpressionSyntax node) =>
-        node.HasExactlyNArguments(1);
-
-    protected override bool IsValueEquality(InvocationExpressionSyntax node, SemanticModel model) =>
-        node.ArgumentList.Arguments[0].GetExpression() is SingleLineLambdaExpressionSyntax lambda
+    protected override bool IsSimpleEqualityCheck(InvocationExpressionSyntax node, SemanticModel model) =>
+        GetArgumentExpression(node, 0) is SingleLineLambdaExpressionSyntax lambda
         && lambda.SubOrFunctionHeader.ParameterList.Parameters is { Count: 1 } parameters
         && parameters[0].Identifier.GetName() is var lambdaVariableName
         && lambda.Body switch
```

Comment:
There is one more condition you need to take care of in VB.NET:

```vb
Dim collection As HashSet(Of String) = New HashSet(Of String)
collection.Add(Nothing)

Dim good = collection.Contains(Nothing)
Dim bad = collection.Any(Function(x) x Is Nothing)

Console.WriteLine(good) ' True
Console.WriteLine(bad) ' True
```

...same logic as `null` checks in C#, different AST. Which means you should track `IsExpression` as well. Some ideas:

```csharp
BinaryExpressionSyntax binary when binary.OperatorToken.IsAnyKind(SyntaxKind.EqualsExpression) =>
    // Check_Equals
BinaryExpressionSyntax binary when binary.OperatorToken.IsAnyKind(SyntaxKind.IsExpression) =>
    // Check_IsNothing
```

...and then the IsNothing check should be similar with the binary logic, only even simpler since you are trying to track _exactly_ the `NothingLiteralExpression`:

![image](https://github.com/SonarSource/sonar-dotnet/assets/115458417/c4655c38-a3a3-4d12-b4e7-ca6208cf81ee)